service/raft/raft_group0.cc

```diff
@@ -413,26 +414,51 @@ future<group0_info> persistent_discovery::run(
     }
 }
 
-future<> raft_group0::abort() {
-    if (_aborted) {
-        co_return;
+future<> raft_group0::abort_and_drain() {
+    if (!_aborted) {
+        // Async lambdas are destroyed at the first co_await,
+        // so accessing lambda-local state (like 'this') afterward would result
+        // in use-after-free. To avoid that, we delegate to a helper function,
+        // do_abort_and_drain().
+
+        _aborted = futurize_invoke([this]() { return do_abort_and_drain(); });
```

Comment:
```suggestion
        _aborted = do_abort_and_drain();
```

Reply:
this wouldn't capture exceptions thrown before the first co_await, would it?

Reply:
You're right. I didn't realize you were thinking about exceptions here. Nevermind.